player/sub.c

```diff
@@ -224,10 +224,18 @@ void reinit_sub(struct MPContext *mpctx, struct track *track)
     osd_set_sub(mpctx->osd, order, track->d_sub);
 
     // When paused we have to wait for packets to be available.
-    // So just retry until we get a packet in this case.
-    if (mpctx->playback_initialized)
-        while (!update_subtitles(mpctx, mpctx->playback_pts) &&
-               mpctx->paused && !mpctx->paused_for_cache);
+    // Retry on a timeout until we get a packet. If still not successful,
+    // then queue it for later in the playloop (but this will have a delay).
+    if (mpctx->playback_initialized) {
+        track->packets_read = false;
+        int64_t end = mp_time_ns() + MP_TIME_MS_TO_NS(50);
+        while (!track->packets_read && mp_time_ns() < end)
+            track->packets_read = update_subtitles(mpctx, mpctx->playback_pts) ||
+                                  !mpctx->paused;
+        if (!track->packets_read)
```

Comment:
what if we are at the end of the file and there is no more subtitle packet to read? It would keep looping?

Reply:
Should already be covered by `handle_update_subtitles`.

Reply:
What if `mpctx->video_status != STATUS_EOF`? Imagine we have 2h video and last subtitle was at minute 30?

Reply:
I'm pretty sure that case returns true because if it didn't then master would currently block forever. But let me make a test file and double check in a bit.

Reply:
Yeah from some testing with some sample files (external subs and soft subs), it always always returns true. The loop that calls `demux_read_packet_async_until` will always end return -1 which ultimately results in this being true. I guess `packets_read` isn't necessarily the best name here.

Reply:
Ok

Reply:
Changed it to `demuxer_ready` instead.